src/python/hooks/file_name_conventions.py

```diff
@@ -0,0 +1,210 @@
+#!/usr/bin/env python3
+# cspell:words colorama, LIGHTRED, LIGHTBLACK, LIGHTWHITE, LIGHTGREEN
+import os
+import re
+import sys
+from pathlib import Path
+
+import yaml
+from colorama import Fore, init
+
+import utils
+
+KEBAB_CASE = r"([a-z]+[a-z0-9\-]*)?"
+
+CASE_REGEXES = {
+    "camelCase": r"^([a-z]+[a-zA-Z0-9]*)?(\.\w+)?$",
+    "snake_case": r"^(_*[a-z]+[a-z0-9_]*)?(\.\w+)?$",
+    "kebab-case": rf"^{KEBAB_CASE}(\.\w+)?$",
+    "PascalCase": r"^([A-Z]+[a-zA-Z0-9]*)?(\.\w+)?$",
+    "UPPER_CASE": r"^(_*[A-Z]+[A-Z0-9_]*)?(\.\w+)?$",
+    "*": r"^.*$",
+}
+
+ERROR_STRING = Fore.RED + "ERROR:" + Fore.RESET
+WARNING_STRING = Fore.LIGHTRED_EX + "WARNING:" + Fore.RESET
+
+root = utils.get_git_root()
+
+
+def load_config(p: Path) -> dict:
+    if p.exists():
+        with open(p, "r") as file:
+            return dict(yaml.safe_load(file))
+    else:
+        print(f"{ERROR_STRING} {p} not found.")
+        sys.exit(1)
+    return {}
```

Comment:
`pylance` is saying that code is unreachable

which means that this might require a new linter?